tests/devices/unit_tests/test_focusing_mirror.py

```diff
@@ -12,121 +16,194 @@
     MirrorVoltageDevice,
     VFMMirrorVoltages,
 )
+from dodal.log import LOGGER
 
 
 @pytest.fixture
 def vfm_mirror_voltages_not_ok(vfm_mirror_voltages) -> VFMMirrorVoltages:
-    vfm_mirror_voltages._channel14_voltage_device._demand_accepted.sim_put(
-        MirrorVoltageDemand.FAIL
+    set_mock_value(
+        vfm_mirror_voltages.voltage_channels[0]._demand_accepted,
+        MirrorVoltageDemand.FAIL,
     )
     return vfm_mirror_voltages
 
 
 @pytest.fixture
 def vfm_mirror_voltages_with_set(vfm_mirror_voltages) -> VFMMirrorVoltages:
-    def not_ok_then_ok(_):
-        vfm_mirror_voltages._channel14_voltage_device._demand_accepted.sim_put(
-            MirrorVoltageDemand.SLEW
+    return vfm_mirror_voltages_with_set_to_value(
+        vfm_mirror_voltages, MirrorVoltageDemand.OK
+    )
+
+
+@pytest.fixture
+def vfm_mirror_voltages_with_set_accepted_fail(
+    vfm_mirror_voltages,
+) -> VFMMirrorVoltages:
+    return vfm_mirror_voltages_with_set_to_value(
+        vfm_mirror_voltages, MirrorVoltageDemand.FAIL
+    )
+
+
+def vfm_mirror_voltages_with_set_to_value(
+    vfm_mirror_voltages, new_value: MirrorVoltageDemand
+) -> VFMMirrorVoltages:
+    async def set_demand_accepted_after_delay():
+        await asyncio.sleep(0.1)
+        set_mock_value(
+            vfm_mirror_voltages.voltage_channels[0]._demand_accepted,
+            new_value,
         )
-        Timer(
-            0.1,
-            lambda: vfm_mirror_voltages._channel14_voltage_device._demand_accepted.sim_put(
-                MirrorVoltageDemand.OK
-            ),
-        ).start()
+        LOGGER.debug("DEMAND ACCEPTED OK")
+
+    def not_ok_then_other_value(_):
+        set_mock_value(
+            vfm_mirror_voltages.voltage_channels[0]._demand_accepted,
+            MirrorVoltageDemand.SLEW,
+        )
+        asyncio.create_task(set_demand_accepted_after_delay())
         return DEFAULT
 
-    vfm_mirror_voltages._channel14_voltage_device._setpoint_v.set = MagicMock(
-        side_effect=not_ok_then_ok
+    vfm_mirror_voltages.voltage_channels[0]._setpoint_v.set = AsyncMock(
+        side_effect=not_ok_then_other_value
     )
-    vfm_mirror_voltages._channel14_voltage_device._demand_accepted.sim_put(
-        MirrorVoltageDemand.OK
+    set_mock_value(
+        vfm_mirror_voltages.voltage_channels[0]._demand_accepted,
+        MirrorVoltageDemand.OK,
     )
     return vfm_mirror_voltages
 
 
 @pytest.fixture
 def vfm_mirror_voltages_with_set_timing_out(vfm_mirror_voltages) -> VFMMirrorVoltages:
     def not_ok(_):
-        vfm_mirror_voltages._channel14_voltage_device._demand_accepted.sim_put(
-            MirrorVoltageDemand.SLEW
+        set_mock_value(
+            vfm_mirror_voltages.voltage_channels[0]._demand_accepted,
+            MirrorVoltageDemand.SLEW,
         )
         return DEFAULT
 
-    vfm_mirror_voltages._channel14_voltage_device._setpoint_v.set = MagicMock(
+    vfm_mirror_voltages.voltage_channels[0]._setpoint_v.set = AsyncMock(
```

Comment:
Should: So `set` is already backed by a mock. you can do:
```python
from ophyd_async.core import get_mock_put
mock = get_mock_put(vfm_mirror_voltages.voltage_channels[0]._setpoint_v)
```
`mock` will already be a mock that you can play with and it should fix your type hinting later on